### PR TITLE
feat(j-s): Indictment Count Legal Arguments

### PR DESCRIPTION
**Pull request by gudjong**
# Indictment Count Legal Arguments

[Heimfærsla - umferðarlagabrot](https://app.asana.com/0/1199153462262248/1203818100039935/f)

## What

- Adds legal arguments to indictment count.
- Generates legal arguments when laws broken change.
- Also adds validation to indictment introduction.

## Why

- Part of traffic violation indictments.

## Screenshots / Gifs

image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx [hunk 1]
```diff
@@ -61,6 +61,42 @@ function lawLabel(law: number[]): string {
...
+  }
+  return 0
+}
+
+function legalArguments(lawsBroken?: number[][] | null) {
+  if (!lawsBroken || lawsBroken.length === 0) {
+    return ''
+  }
+
+  let legalArguments = `Telst háttsemi þessi varða við ${lawsBroken[0][1]}. mgr.`
```
Comment (unakb) [inline apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx:85]: According to the design the "mgr." text should only come after the last law in a list that falls under the same gr., 

So the code would generate this:
Telst háttsemi þessi varða við 1. mgr., sbr. 2. mgr. 48. gr., sbr. 1. mgr. 95. gr. umferðarlaga nr. 77/2019.

But it should be like this:
Telst háttsemi ákærða varða við 1., sbr. 2. mgr. 50. gr., sbr. 1. mgr. 95. gr. umferðarlaga nr. 77/2019.

File: apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx [hunk 2]
```diff
@@ -61,6 +61,42 @@ function lawLabel(law: number[]): string {
...
+    if (lawsBroken[i][0] !== lawsBroken[i - 1][0]) {
+      legalArguments = `${legalArguments} ${lawsBroken[i - 1][0]}. gr.`
+    }
+
+    legalArguments = `${legalArguments}, sbr. ${lawsBroken[i][1]}. mgr.`
+  }
+
+  return `${legalArguments} ${
+    lawsBroken[lawsBroken.length - 1][0]
+  }. gr. umferðarlaga nr. 77/2019.`
```
Comment (unakb) [inline apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx:97]: Should we put these hardcoded parts of the strings in to Contentful ?

**Review by unakb — CHANGES_REQUESTED** [review]
Just the minor string fix for the combined legal arguments, otherwise looks good! ☺️

File: apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx:85]: Well spotted, will fix.

File: apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx (hunk 2 above)
Comment (gudjong) [reply apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx:97]: But, of course we should 😊 Will fix.

**Review by unakb — APPROVED** [review]
Looks good!

**Pull request merged**